Update changelog for bytestring-0.12.2.0

Bodigrim opened the pull request.

clyring commented:
The existing entries and organization look good to me.

* #680 also adds `NFData` instances for `BufferRange` and `Buffer`.
* #672 also changes `fromListN` in the `IsList` instance for `StrictByteString`.
  * Technically this is a breaking change as-is, since the use of `packUptoLenBytes` means that the input list will be truncated if the list is longer than the size hint suggests.
* #672 also moves `toLazyByteString` into `Data.ByteString.Builder.Internal` and exports it there.

Also, we are still only working around #665 in a Hackage revision. We should fix it properly (or at least merge the contents of the Hackage revision into the source repo) before cutting a release. I can prepare a MR for this tomorrow, if you don't beat me to it.

Bodigrim commented:
> Technically this is a breaking change as-is, since the use of `packUptoLenBytes` means that the input list will be truncated if the list is longer than the size hint suggests.

Yeah, maybe it's better to revert the change to `instance IsList StrictByteString`.

> We should fix it properly (or at least merge the contents of the Hackage revision into the source repo)

Let's make our lives easier and just backport the revision.

----

Feel free to push any corrections into this branch.

Bodigrim commented:
Updated now, I reverted changes to `instance IsList ByteString` in `bytestring-0.12` branch.

@clyring we are being chased by GHC team to make a release asap. If there are no more comments by Friday evening, I'll merge this and release `bytestring-0.12.2.0`.

clyring reviewed (approved):
LGTM. I went ahead and added a few (invisible) entries for the last backports.

The commit message for 537b8ac7389a0f5dbd1e4e4e4afdf9f44f5e06a2 unfortunately refers to the wrong PR.

Thanks much for working on this! Sorry for disappearing on you these past few weeks.
